docs/workflows/accessing-payload-data.md

```diff
@@ -0,0 +1,330 @@
+# Accessing Payload Data
+
+There are a few different methods for accessing and manipulating data from workflow payloads and device data queries. Depending on the use case, you may run across a point while building your app where you're asked to supply a data reference in one of the following formats.
+
+```json
+{
+  "data" : {
+    "aNumberProp": 42,
+    "aStringProp": "Hello world",
+    "aBooleanProp": true,
+    "aFalsyProp" : 0,
+    "aDecimal": 123.456789,
+    "anUndefinedProp": undefined,
+    "aNullProp": null,
+    "aJavascriptDateObject": Wed Sep 28 2016 15:40:25 GMT-0400 (EDT),
+    "anArray": [
+      44,
+      "Goodbye world",
+      false,
+      {
+        "anObjectPropInAnArray": "I'm deep!"
+      }
+    ],
+    "anObject": {
+      "aNestedArray": [
+        "World Star!",
+        20,
+        { }
+      ],
+      "aNestedNumber": 56,
+      "aNestedString": "Wally World"
+    }
+  }
+}
+```
+
+## Payload Paths
+
+Many workflow nodes, such as the [HTTP node](/workflows/data/http/), the [Get](/workflows/data/get-value/) and [Store Value](/workflows/data/store-value/) nodes and the [Math node](/workflows/logic/math/), require users to define a payload path at which either data will be accessed for evaluation, or at which the returned result of an operation will be stored. In these cases we use **payload paths**.
+
+Payload paths are dot-separated references to object properties, such as `foo.bar`. If any property in the chain returns `undefined`, the value of the payload path will also return undefined.
+
+When accessing an array value by index, the index must be wrapped in square brackets and preceded by a dot, such as  `foo.bar.[1]`. Square brackets can also be used to access specific characters in a string value.
+
+### Examples
+
+Given our example object, here is how we would reference many of the properties using payload paths:
+
+```javascript
+data.aNumberProp // 42
+data.aBooleanProp // false
+data.aNullProp // null
+data.aJavascriptDateObject // Wed Sep 28 2016 15:40:25 GMT-0400 (EDT)
+
+data.anArray.[1] // "Goodbye world"
+data.anArray.1 // (undefined. This is looking for the property name "1" on the object "anArray".)
+data.anArray.[1].[2] // o (The character at index 2 of the array item at index 1)
+data.anArray.[3] // { "anObjectPropInAnArray": "I'm deep!" }
+data.anArray.[3].anObjectPropInAnArray // "I'm deep!"
+data.anArray.[8] // (undefined. This index does not exist on the array.)
+
+data.anObject.aNestedNumber // 56
+data.anObject.aNestedArray // ["World Star!", 20, { }]
+data.anObject.aNestedArray.[1] // 20
+
+data.aNonexistentObject.aMissingProp // (undefined. "aNonExistentObject" is not a property of "data")
+aNumberProp // (undefined. This is not a top-level property of our object.)
+```
+
+Payload paths **do not accept variables** within the path definition; the paths must be static references.
+
+## String Templates
```

Comment:
general statement and link that we are using handlebars here at the top of this section?